sound/soc/sof/intel/hda.c

```diff
@@ -1216,17 +1219,9 @@ static bool link_slaves_found(struct snd_sof_dev *sdev,
 			/* find out how many identical parts are expected */
 			for (k = 0; k < link->num_adr; k++) {
 				u64 adr2 = link->adr_d[k].adr;
-				unsigned int part_id2, link_id2, mfg_id2, version2;
-
-				mfg_id2 = SDW_MFG_ID(adr2);
-				part_id2 = SDW_PART_ID(adr2);
-				link_id2 = SDW_DISCO_LINK_ID(adr2);
-				version2 = SDW_VERSION(adr2);
 
-				if (link_id2 == link_id &&
-				    part_id2 == part_id &&
-				    mfg_id2 == mfg_id &&
-				    version2 == version)
+				if (SDW_UNIQUE_ID_STRIPPED_ADR(adr2) ==
+				    SDW_UNIQUE_ID_STRIPPED_ADR(adr))
```

Comment:
this re-adds the problem you had before, now the class information is taken into account, and it may or may not be  identical.

I think it's better to stick to an explicit list of fields that are taken into account, or you have to extend your macro to remove the class information.

Reply:
> this re-adds the problem you had before, now the class information is taken into account, and it may or may not be identical.
> 
> I think it's better to stick to an explicit list of fields that are taken into account, or you have to extend your macro to remove the class information.

Thanks @plbossart I didn't notice that the class id can be different. I will modify the macro to fix it.